blockchain-agent/src/main/java/com/sphereon/alfresco/blockchain/agent/config/RestControllerConfigTemplate.java

```diff
@@ -236,193 +230,167 @@ public ObjectMapper primaryObjectMapper() {
+    RestTemplate restTemplate(@Qualifier("mappingJackson2HttpMessageConverter") final MappingJackson2HttpMessageConverter httpMessageConverter) {
         RestTemplate restTemplate = new RestTemplate();
         restTemplate.getMessageConverters().add(httpMessageConverter);
         return restTemplate;
     }
 
-
     public interface SimplifiedDocketConfigurator {
-
         void configureDocket(Builder docketBuilder, Mode mode);
 
         void configureApiInfo(ApiInfoBuilder apiInfoBuilder, Mode mode);
 
-
         final class Builder {
             final SphereonDocketConfig sphereonDocketConfig = new SphereonDocketConfig();
 
-
             public Builder withGatewayHostname(String gatewayHostName) {
                 sphereonDocketConfig.setGatewayHostName(gatewayHostName);
                 return this;
             }
 
-
             public Builder withPathMapping(Mode mode, String pathMapping) {
 
                 sphereonDocketConfig.setPathMapping(mode, pathMapping);
                 return this;
             }
 
-
             public Builder withApiSelector(Predicate<RequestHandler> apiSelector) {
                 sphereonDocketConfig.setApiSelector(apiSelector);
                 return this;
             }
 
-
             public Builder withApiSelector(String basePackage) {
                 sphereonDocketConfig.setApiSelector(RequestHandlerSelectors.basePackage(basePackage));
                 return this;
             }
 
-
             public Builder withPathSelector(Predicate<String> selector) {
                 sphereonDocketConfig.setPathSelector(selector);
                 return this;
             }
 
-
             public Builder withPathSelector(String regexSelector) {
                 sphereonDocketConfig.setPathSelector(PathSelectors.regex(regexSelector));
                 return this;
             }
 
-
             public Builder withTags(Tag first, Tag... remaining) {
                 sphereonDocketConfig.firstTag = first;
                 sphereonDocketConfig.additionalTags.addAll(Sets.newHashSet(remaining));
                 return this;
             }
 
-
             public SphereonDocketConfig build() {
                 return sphereonDocketConfig;
             }
         }
     }
 
-
     private static class SphereonDocketConfig {
-
         @Value("${sphereon.api.gateway-address:gw.api.cloud.sphereon.com}")
         private String gatewayHostName;
 
-        private final Map<Mode, String> pathMappings = new HashMap<>();
+        private final Map<Mode, String> pathMappings;
 
-        private final Set<Tag> additionalTags = Sets.newHashSet();
+        private final Set<Tag> additionalTags;
 
         private Tag firstTag;
 
-        private Predicate<RequestHandler> apiSelector = RequestHandlerSelectors.any();
+        private Predicate<RequestHandler> apiSelector;
 
         private Predicate<String> pathSelector;
 
+        private SphereonDocketConfig() {
+            pathMappings = new HashMap<>();
```

Comment:
assignement -> this.

Reply:
Done. Any reason not to do the same for other usages?